src/common/transformations/src/transformations/op_conversions/convert_maxpool_downgrade.cpp

```diff
@@ -53,6 +50,43 @@ pass::ConvertMaxPool8ToMaxPool1::ConvertMaxPool8ToMaxPool1() {
         return true;
     };
 
-    auto m = make_shared<pattern::Matcher>(maxpool_v8_pattern, matcher_name);
+    auto m = std::make_shared<pattern::Matcher>(maxpool_v8_pattern, matcher_name);
+    register_matcher(m, callback);
+}
+
+ov::pass::ConvertMaxPool14ToMaxPool8::ConvertMaxPool14ToMaxPool8() {
+    MATCHER_SCOPE(ConvertMaxPool14ToMaxPool8);
+
+    const auto max_pool_v14_pattern = pattern::wrap_type<ov::op::v14::MaxPool>();
+
+    const matcher_pass_callback callback = [](pattern::Matcher& m) {
+        const auto max_pool_v14 = std::dynamic_pointer_cast<ov::op::v14::MaxPool>(m.get_match_root());
+        if (!max_pool_v14) {
+            return false;
+        }
+
+        const auto rounding_type_v14 = max_pool_v14->get_rounding_type();
+        const auto rounding_type_v8 =
```

Comment:
This check for rounding is same  for avg pool and looks similar in shape inference.
Maybe we can have function in shape infer for it and just use it here?

Reply:
The check has been refactored when introducing workaround subgraph and is no longer there.